packages/runtime/datastore/src/channelDeltaConnection.ts

```diff
@@ -36,7 +37,18 @@ export class ChannelDeltaConnection implements IDeltaConnection {
     }
 
     public process(message: ISequencedDocumentMessage, local: boolean, localOpMetadata: unknown) {
-        this.handler.process(message, local, localOpMetadata);
+        try {
```

Comment:
Couple questions:
1. Is this required? I.e. we already should catch exceptions thrown from here at DeltaQueue level and propagate them to Container.close(). Is this just a safety net for the cases where it does not happen? Why here, and not on some other level?
2. We could (and likely should) add all these properties in Container.close directly to any error that reaches container.close. Consider the case where I roll alternative implementation to DataStoreRuntime that does not use this code - we will not get benefits of code in here.

Reply:
I'm looking over all this infrastructure again and I'm realizing that the understanding of the flow of ops was incomplete at best...

Am I correct (now) that:
1. _pretty much every_ op flows through `IFluidDataStoreChannel.process()`?
2. *most* ops are going through `IFluidDataStoreChannel` concrete implementation instances constructed by a `DataObjectFactory` (and the concrete type of `IFluidDataStoreChannel` is parameterized with an argument to the `DataObjectFactory` constructor)?

If those are the cases, perhaps the "right level" is to handle this outside of the framework repo, at call-sites that construct `DataObjectFactory` instances -- much like it seems like it seems `mixinRequestHandler` and `mixinSummaryHandler` suggest.

That said, providing a capability for a consumer of the framework to intercept/guard (potentially async/pending) ops passed to `LocalChannelContext` and/or `RemoteChannelContext` instances would be appreciated.  Would you entertain the idea of parameterizing these types/constructors through `FluidDataStoreRuntime` construction -- thus enabling similar sorts of lighter weight "mixin" customizations?